core/src/main/scala/org/apache/spark/sql/tispark/BasicExpression.scala

```diff
@@ -130,6 +137,20 @@ object BasicExpression {
       case Like(BasicExpression(lhs), BasicExpression(rhs)) =>
         Some(StringRegExpression.like(lhs, rhs))
 
+      // Coprocessor has its own behavior of type promoting and overflow check
+      // so we simply remove it from expression and let cop handle it
+      case CheckOverflow(BasicExpression(expr), _) =>
```

Comment:
should we remove MakeDecimal and UnscaledValue?

Reply:
see discussions above....

Reply:
ok.